File: algorithm/#0080/code.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        l=len(nums)
        i=1
        cnt=1
        while i<l :
            if nums[i]!=nums[i-1] :
                cnt=1
            else :
                cnt+=1
            if cnt>2 :
                del nums[i]
                l-=1
            else :
                i+=1

File: algorithm/#0026/code.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int n=nums.size();
        if(n==0) return 0;
        int p1=1, p2=1;
        while(p1<n) {
            if(nums[p1]!=nums[p1-1]) {
                nums[p2]=nums[p1];
                ++p2;
            }
            ++p1;
        }
        return p2;
    }
};
